File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int q = 0; //initialize variables
    int nick = 0;
    int d = 0;
    int p = 0;
    float x;

    do
    {
        x = get_float("Change owed: "); //get change in dollars
    }
    while (x < 0); //repromt user if change is less than 0

    int n = round(x * 100); //convert change to cents and round to nearest penny

    while (n >= 25)
    {
        q = n / 25; //figure out how many quarters fit in the sum
        n = n - (q * 25); //update value of n (remaining change that needs to be sorted)
    }

    while (n >= 10)
    {
        d = n / 10; //figure out how many dimes fit in the sum
        n = n - (d * 10); //update value of n
    }

    while (n >= 5)
    {
        nick = n / 5; //figure out how many nickels fit in the sum
        n = n - (nick * 5); //update value of n
    }

    while (n >= 1)
    {
        p = n; //remaining change can be sorted into pennies
        n = 0; //update final value of n (all change has been sorted)
    }

    int coin = q + nick + d + p; //add up total number of coins used

    printf("%i\n", coin); //output coins used
}

File: pset6/cash/cash.py
from cs50 import get_float
import sys

while True:  # get change amount
    change = get_float("Change owed: ")
    change = int(change * 100)  # convert to cents

    if change >= 0:  # positive value received
        break

coins_val = [25, 10, 5, 1]  # coin values
total_coins = 0  # total number of coins used in change

for i in range(5):  # greedy algorithm

    if change == 0:  # change amount has been paid
        print(total_coins)  # print number of coins used
        sys.exit()

    num_coins = int(change / coins_val[i])  # check how many of the current coin [i] fits into change
    total_coins += num_coins  # update total coins use
    change = change - (num_coins * coins_val[i])  # update remaining change value